datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -96,42 +94,23 @@ public SdxRecoverableResponse validateRecovery(SdxCluster sdxCluster) {
-        SdxStatusEntity actualStatusForSdx = sdxStatusService.getActualStatusForSdx(sdxCluster);
-        if (entitlementService.isDatalakeResizeRecoveryEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
-            switch (actualStatusForSdx.getStatus()) {
-                case STOP_FAILED:
-                    return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxStartFlow(sdxCluster));
-                case STOPPED:
-                    return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxResizeRecovery(sdxCluster, null));
-                case PROVISIONING_FAILED:
-                    return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxResizeRecovery(getOldClusterErrorIfNotFound(sdxCluster), sdxCluster));
-                case RUNNING:
-                    if (actualStatusForSdx.getStatusReason().contains("Datalake restore failed")) {
-                        return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxResizeRecovery(getOldClusterErrorIfNotFound(sdxCluster), sdxCluster));
-                    } else {
-                        throw new NotImplementedException("Cluster is currently running and cannot be recovered");
-                    }
-                default:
-                    throw new NotImplementedException("Cluster is currently in an unrecoverable state");
-            }
-        } else {
-            throw new BadRequestException("Entitlement for resize recovery is missing");
+        Optional<SdxCluster> oldCluster = getOldCluster(sdxCluster);
+        if (oldCluster.isPresent()) {
+            return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxResizeRecovery(oldCluster.get(), sdxCluster));
         }
-    }
-
-    private SdxCluster getOldClusterErrorIfNotFound(SdxCluster newCluster) {
-        return getOldCluster(newCluster).orElseThrow(notFound(
-                "detached SDX cluster",
-                "Env CRN: " + newCluster.getEnvCrn() + ", Account ID: " + newCluster.getAccountId()
-        ));
+        return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxResizeRecovery(sdxCluster, null));
     }
 
     private Optional<SdxCluster> getOldCluster(SdxCluster newCluster) {
-        Optional<SdxCluster> oldCluster = sdxClusterRepository.findByAccountIdAndEnvCrnAndDeletedIsNullAndDetachedIsTrue(
+        List<SdxCluster> clusters = sdxClusterRepository.findByAccountIdAndEnvCrnAndDeletedIsNull(
                 newCluster.getAccountId(), newCluster.getEnvCrn()
         );
-        return (oldCluster.isPresent() && !oldCluster.get().getClusterName().equals(newCluster.getClusterName())) ?
-                oldCluster : Optional.empty();
+        if (clusters.size() > 2) {
+            LOGGER.warn("Found more than 2 clusters from environment! This should not normally be possible.");
+        }
+        return clusters.stream()
+                .filter(cluster -> !cluster.getId().equals(newCluster.getId()))
+                .findAny();
     }
```

Comment:
What is the functional change you are adding with this change?

Reply:
No longer necessary, I have pulled this change back.